src/jupyterlabcontroller/services/lab.py

```diff
@@ -686,27 +687,29 @@ def build_volumes(self, username: str) -> List[LabVolumeContainer]:
         vols.append(runtime_vol)
         return vols
 
-    def build_init_ctrs(self, username: str) -> List[V1Container]:
+    def build_init_ctrs(
+        self, username: str, resources: UserResources
+    ) -> List[V1Container]:
         init_ctrs: List[V1Container] = []
         ic_volumes: List[LabVolumeContainer] = []
         for ic in self.lab_config.init_containers:
             if ic.volumes is not None:
                 ic_volumes = self.build_lab_config_volumes(ic.volumes)
             ic_vol_mounts = [x.volume_mount for x in ic_volumes]
-            ic_sec_ctx = (
-                V1SecurityContext(
-                    run_as_non_root=True,
-                    run_as_user=1000,
-                    allow_privilege_escalation=False,
-                ),
-            )
             if ic.privileged:
                 ic_sec_ctx = V1SecurityContext(
                     run_as_non_root=False,
                     run_as_user=0,
                     allow_privilege_escalation=True,
                 )
+            else:
+                ic_sec_ctx = V1SecurityContext(
+                    run_as_non_root=True,
+                    run_as_user=1000,
```

Comment:
Is there ever going to be a case when we want to run an unprivileged user container but we want to run it as the right user?  I couldn't think of one, because I think whatever it is we want to do, we probably want to do in the lab, but I could be convinced that we should plug the real uid in here.

Reply:
Yeah, I think we probably should default to running init containers as the user. This was pre-existing, so I'll add that to my to-do list for later attention.